src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/SqlCommand/SqlCommandCancelTest.cs

```diff
@@ -558,5 +573,46 @@ private static async Task AsyncCancelDoesNotWait(string connStr)
                 Assert.InRange((ended - started).TotalSeconds, cancelSeconds, delaySeconds - 1);
             }
         }
+
+        private static void AsyncCancelDoesNotWait2(string connStr)
+        {
+            const int delaySeconds = 30;
+            const int cancelSeconds = 1;
+
+            var cancellationTokenSource = new CancellationTokenSource();
+            DateTime started = DateTime.UtcNow;
+            DateTime ended = DateTime.UtcNow;
+            Exception exception = null;
+
+            Task executing = ExecuteWaitForAsync(cancellationTokenSource.Token, connStr, delaySeconds);
+
+            cancellationTokenSource.CancelAfter(TimeSpan.FromSeconds(cancelSeconds+1));
+
+            try
+            {
+                executing.Wait();
+            }
+            catch (Exception ex)
+            {
+                exception = ex;
+            }
+            ended = DateTime.UtcNow;
+
+            Assert.NotNull(exception);
+            Assert.InRange((ended - started).TotalSeconds, cancelSeconds, delaySeconds - 1);
+        }
+
+        private static async Task ExecuteWaitForAsync(CancellationToken cancellationToken, string connectionString, int delaySeconds)
+        {
+            using (var connection = new SqlConnection(connectionString))
+            {
+                await connection.OpenAsync().ConfigureAwait(false);
+                using (var command = new SqlCommand($"WAITFOR DELAY '00:00:{delaySeconds:D2}'", connection))
```

Comment:
When using WAITFOR DELAY, we cannot reproduce this issue, are you sure this test failed without your changes?

Reply:
No. I did the fix dev using the original repro from the report and then test-ified the repro after so that it used wait which I thought acted the same way. I've changed to the loop from the repro in the test now. Sorry about that.